File: shoring/flywheel/src/main/java/com/panopset/flywheel/TemplateDirectiveCommand.java
package com.panopset.flywheel;

import java.io.StringWriter;
import com.panopset.compat.Stringop;

/**
 * Command that is created from a directive.
 */
public class TemplateDirectiveCommand extends TemplateCommand {

  public TemplateDirectiveCommand(final TemplateLine templateLine, final String innerPiece,
      final Template template) {
    super(templateLine, innerPiece, template);
    String params = "";
    if (innerPiece.length() > 2) {
      params = innerPiece.substring(Syntax.getDirective().length() + 2);
    }
    
    String replacement = getTemplate().getFlywheel().get(params);
    
    if (Stringop.isPopulated(replacement)) {
      setParams(replacement);
    } else {
      setParams(params);
    }
  }

  @Override
  public void resolve(final StringWriter sw) throws FlywheelException {
    // does nothing here.
  }

}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/TemplateDirectiveCommand.kt
package com.panopset.flywheel

import com.panopset.compat.Stringop
import java.io.StringWriter

/**
 * Command that is created from a directive.
 */
open class TemplateDirectiveCommand(
    templateLine: TemplateLine, innerPiece: String,
    template: Template
) : TemplateCommand(templateLine, innerPiece, template) {


    init {
        var params = ""
        if (innerPiece.length > 2) {
            params = innerPiece.substring(Syntax.getDirective().length + 2)
        }
//        val replacement = template.flywheel.getEntry(params)
//        parms = if (Stringop.isPopulated(replacement)) {
//            replacement
//        } else {
//            params
//        }
        parms = params
    }

    override fun resolve(sw: StringWriter) {
        // does nothing here.
    }

    fun mapValueFirstThenExplicit(params: String): String {
        val replacementFileName = template.flywheel.getEntry(params)
        return if (Stringop.isPopulated(replacementFileName)) {
            replacementFileName
        } else params
    }
}
